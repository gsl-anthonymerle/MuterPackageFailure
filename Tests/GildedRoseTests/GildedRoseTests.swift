@testable import GildedRose
import XCTest
import ApprovalTests_Swift

class GildedRoseTests: XCTestCase {
    fileprivate func doUpdateQuality(_ name: String, _ sellIn: Int, _ quality: Int) -> String {
        let items = [Item(name: name, sellIn: sellIn, quality: quality)]
        let app = GildedRose(items: items)
        
        app.updateQuality()

        return items[0].description
    }

    func testItems() throws {
        try CombinationApprovals.verifyAllCombinations(
            doUpdateQuality(_:_:_:),
            ["foo", "Aged Brie", "Backstage passes to a TAFKAL80ETC concert"],
            [0],
            [0, 1, 2]
        )
    }
}
