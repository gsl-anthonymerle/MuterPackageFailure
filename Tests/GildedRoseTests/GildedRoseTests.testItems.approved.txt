["foo", 0, 0] -> foo, -1, 0
["foo", 0, 1] -> foo, -1, 0
["foo", 0, 2] -> foo, -1, 0
["Aged Brie", 0, 0] -> Aged Brie, -1, 2
["Aged Brie", 0, 1] -> Aged Brie, -1, 3
["Aged Brie", 0, 2] -> Aged Brie, -1, 4
["Backstage passes to a TAFKAL80ETC concert", 0, 0] -> Backstage passes to a TAFKAL80ETC concert, -1, 0
["Backstage passes to a TAFKAL80ETC concert", 0, 1] -> Backstage passes to a TAFKAL80ETC concert, -1, 0
["Backstage passes to a TAFKAL80ETC concert", 0, 2] -> Backstage passes to a TAFKAL80ETC concert, -1, 0
